### PR TITLE
INT-560_add badge on product page and cart

### PR DESCRIPTION
**Pull request by Francois-Gomis**

### CODE REVIEW

**Review by syjust-alma — COMMENTED** [review]
[sonarcloud](https://sonarcloud.io/project/issues?id=alma_alma-installments-magento1&pullRequest=6&resolved=false&types=CODE_SMELL):

* about the class naming convention: I edited the quality profile used by the Magento 1 project to avoid this issue like in the wordpress project
* about the [too many methods in the class](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=6&id=alma_alma-installments-magento1&open=AYCE_pEhE6qqalXZMkaL): please can you split the class with an other logical helper for exemple ?
* about security hubspot, I will discuss with security committee to get an external opinion.

I will do a proofest review when there's no more sonar issues

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php [hunk 1]
```diff
@@ -322,4 +328,32 @@ public function validateFeePlanMinAndMaxCustomAmount($feePlans)
...
 
+
+    /**
+     * @return string
+     */
+    public function getEnableFeePlansForBadge()
+    {
+        $plansForBadge = $this->getEnabledFeePlansConfigFromBackOffice();
+        foreach ($plansForBadge as $panKey => $plan) {
+            $plansForBadge[$panKey] = $this->formatPlanForBadge($plan);
```
Comment (PaulRouss3l) [inline src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: I see that later you use `array_values` to remove the keys.
Why don't you use another array ?
```
$formattedBadgePlans = [];
foreach ($plansForBadge as $plan) {
    $formattedBadgePlans[] = $this->formatPlanForBadge($plan);
}
return json_encode($formattedBadgePlans);
```

File: src/app/design/frontend/base/default/template/alma/product/badge.phtml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    /**
+     * @description Change widget position
+     */
+    function moveToCustomPosition(){
+        var customPositionName = '<?= $this->getWidgetCustomPosition()?>';
+        if(customPositionName !== '' ){
+            var customPosition = document.getElementById(customPositionName);
+            if (customPosition == null){
```
Comment (PaulRouss3l) [inline src/app/design/frontend/base/default/template/alma/product/badge.phtml]: unless you add
```
if (customPosition == null && document.getElementsByClassName(customPositionName).lenght > 0){
```
this line : 
```
customPosition = document.getElementsByClassName(customPositionName)[0]
```
will prompt an error

File: src/app/design/frontend/base/default/template/alma/product/badge.phtml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    /**
+     *
+     * @param priceContainer Container with the price to extract
+     * @returns {number} price in cent
+     */
+    function getPriceFromContainer(priceContainer){
+        var price = 0;
+        if(priceContainer !== undefined && priceContainer !== null && priceContainer.innerText !== undefined && priceContainer.innerText !== null)
```
Comment (PaulRouss3l) [inline src/app/design/frontend/base/default/template/alma/product/badge.phtml]:
```
if(priceContainer && 'innerText' in priceContainer && priceContainer.innerText !== null)
```
colle mieux a ce que tu veux faire je pense

File: src/app/design/frontend/base/default/template/alma/product/badge.phtml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            }
+        }
+    }
+
+    /**
+     *  @description Add listener on quantity field.
+     */
+    function addListenerToQty(){
+        var qtyNode = document.getElementById('qty');
+        qtyNode.addEventListener("input",function(){updateWidget()});
```
Comment (PaulRouss3l) [inline src/app/design/frontend/base/default/template/alma/product/badge.phtml]:
```
qtyNode.addEventListener("input", updateWidget);
```
est suffisant ;)

File: src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php (hunk 1 above)
Comment (Francois-Gomis) [reply src/app/code/community/Alma/Installments/Helper/FeePlansHelper.php]: Done

File: src/app/design/frontend/base/default/template/alma/product/badge.phtml (hunk 2 above)
Comment (Francois-Gomis) [reply src/app/design/frontend/base/default/template/alma/product/badge.phtml]: No error with 
`customPosition = document.getElementsByClassName(customPositionName)[0]`
but it's better to protect var

Done

File: src/app/design/frontend/base/default/template/alma/product/badge.phtml (hunk 3 above)
Comment (Francois-Gomis) [reply src/app/design/frontend/base/default/template/alma/product/badge.phtml]: I didn't know "In operator" Thank  :) 
Done

File: src/app/design/frontend/base/default/template/alma/product/badge.phtml (hunk 4 above)
Comment (Francois-Gomis) [reply src/app/design/frontend/base/default/template/alma/product/badge.phtml]: Thank
Same on line 78 ? 
` priceContainer.on('DOMSubtreeModified', function () {
            updateWidget();
        });`
Done

File: src/app/design/frontend/base/default/template/alma/cart/cart-widget.phtml [hunk 5]
```diff
@@ -27,30 +27,37 @@
...
+    <script src="https://cdn.jsdelivr.net/npm/@alma/widgets@2.x/dist/widgets.umd.js"></script>
+    <script>
 
-        .alma-cart-eligibility.widget {
-            padding: 0 20px;
-        }
+        /**
+         * @description  Wait for end of Dom loading / qty is not load immediately
+         */
+        window.addEventListener('DOMContentLoaded', () => {
```
Comment (PaulRouss3l) [inline src/app/design/frontend/base/default/template/alma/cart/cart-widget.phtml]: window.addEventListener('DOMContentLoaded', updateWidget);

**Review by PaulRouss3l — APPROVED** [review]

File: src/app/design/frontend/base/default/template/alma/cart/cart-widget.phtml (hunk 5 above)
Comment (Francois-Gomis) [reply src/app/design/frontend/base/default/template/alma/cart/cart-widget.phtml]: done

**Pull request merged**